batch-models/src/main/scala/org/ekstep/analytics/model/report/CourseEnrollmentModel.scala

```diff
@@ -0,0 +1,98 @@
+package org.ekstep.analytics.model.report
+
+import java.text.SimpleDateFormat
+import java.util.Calendar
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, Encoders, SQLContext}
+import org.ekstep.analytics.framework._
+import org.ekstep.analytics.framework.util.{JSONUtils, JobLogger}
+import org.ekstep.analytics.job.report.{BaseCourseMetrics, BaseCourseMetricsOutput}
+import org.ekstep.analytics.model.ReportConfig
+import org.ekstep.analytics.util.CourseUtils
+import org.sunbird.cloud.storage.conf.AppConf
+
+case class CourseEnrollmentOutput(date: String, courseName: String, batchName: String, status: String, enrollmentCount: BigInt, completionCount: BigInt, slug: String, reportName: String) extends AlgoOutput with Output
+case class ESResponse(participantCount: BigInt, completedCount: BigInt, courseId: String)
+
+object CourseEnrollmentModel extends BaseCourseMetrics[Empty, BaseCourseMetricsOutput, CourseEnrollmentOutput, CourseEnrollmentOutput] with Serializable {
+
+  implicit val className: String = "org.ekstep.analytics.model.CourseEnrollmentModel"
+  override def name: String = "CourseEnrollmentModel"
+
+  override def algorithm(events: RDD[BaseCourseMetricsOutput], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[CourseEnrollmentOutput] = {
+    implicit val sqlContext = new SQLContext(sc)
+    val finalRDD = getCourseEnrollmentOutput(events)
+    val date = (new SimpleDateFormat("dd-MM-yyyy")).format(Calendar.getInstance().getTime)
+    finalRDD.map(f => CourseEnrollmentOutput(date,f._1.courseName,f._1.batchName,f._1.status,
+      f._2.getOrElse(ESResponse(0,0,"")).participantCount,f._2.getOrElse(ESResponse(0,0,"")).completedCount,f._1.slug,"course_enrollments"))
+  }
+
+  override def postProcess(data: RDD[CourseEnrollmentOutput], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[CourseEnrollmentOutput] = {
+    implicit val sqlContext = new SQLContext(sc)
+    if (data.count() > 0) {
+      val configMap = config("reportConfig").asInstanceOf[Map[String, AnyRef]]
+      val reportConfig = JSONUtils.deserialize[ReportConfig](JSONUtils.serialize(configMap))
+
+      import sqlContext.implicits._
+      reportConfig.output.map { f =>
+        if (f.`type`.equals("csv")) {
+          val df = data.toDF().na.fill(0L)
+          val storageService = fc.getStorageService("azure")
+          CourseUtils.postDataToBlob(df, f,config, storageService)
+          storageService.closeContext()
```

Comment:
1. Remove the storage service implementation
2. Do not hardcode it for csv format

Reply:
Done